dspinner/pin.go

```diff
@@ -318,7 +324,13 @@ func (p *pinner) removePin(ctx context.Context, pp *pin) error {
 }
 
 // Unpin a given key
-func (p *pinner) Unpin(ctx context.Context, c cid.Cid, recursive bool) error {
+func (p *pinner) Unpin(ctx context.Context, c cid.Cid, recursive bool) (perr error) {
+	defer func() {
+		if perr == nil {
+			perr = p.Flush(ctx)
+		}
+	}()
```

Comment:
seems fine, although we could probably do this at the end of the function. Either way let's document what we're doing with these flushes (and why) somewhere in this PR

Reply:
Let's leave it like this, since remove sometimes fixes the index without removing and taking care of these cases may be error prone without much value add. Added comment. In https://github.com/ipfs/go-ipfs-pinner/pull/13/commits/a24f5872bea10d592361febcd6bb3bf0ea5743e0.